### PR TITLE
IS-15210: Add update-schemas functionality

### PR DESCRIPTION
**Pull request by hglad**

### CODE REVIEW

File: pyproject.toml
Comment (mMaydew) [inline pyproject.toml:1]: Can we remove these? :sweat_smile:

File: setup.cfg
Comment (mMaydew) [inline setup.cfg:1]: Can we remove these? :sweat_smile:

File: connector_cli/connectorpy.py [hunk 1]
```diff
@@ -169,6 +172,7 @@ def expand_connector(
                 component["permissions"] = endpoint_permissions
                 logger.warning(
                     "Set permissions for endpoint pipe"
+                    f""
```
Comment (mMaydew) [inline connector_cli/connectorpy.py]: What's this for?

File: connector_cli/connectorpy.py [hunk 2]
```diff
@@ -244,7 +248,7 @@ def collapse_connector(
                 should_warn = True
         if should_warn:
             warning_text = (
-                "WARNING! There is no use for template parameter(s) "
+                "WARNING! There is no use for template parameter(s)"
```
Comment (mMaydew) [inline connector_cli/connectorpy.py:250]: The extra space should be here otherwise there is no space, it's linked to the f-string underneath

**Review by tombech — APPROVED** [review]

File: connector_cli/connectorpy.py [hunk 3]
```diff
@@ -305,3 +308,191 @@ def collapse_connector(
...
+                )
+        elif property_type == "array":
+            items_schema = property_schema["items"]
+            if items_schema.get("type", "") == "object":
+                for subproperty_name, subschema in items_schema["properties"].items():
+                    description = subschema.get("description", "").replace('"', '"')
+                    subproperty_type = subschema.get("type")
+                    outfile.write(
+                        f'" {system}","{datatype}","{property_name}",'
+                        f""
```
Comment (mMaydew) [inline connector_cli/connectorpy.py:467]: I'm guessing these are new lines at this point? Why not `\n`

**Review by mMaydew — COMMENTED** [review]
Other than these LGTM

File: connector_cli/connectorpy.py (hunk 2 above)
Comment (hglad) [reply connector_cli/connectorpy.py:250]: I know

**Pull request merged**